### PR TITLE
Improve AppBar Customization

### PR DESCRIPTION
**Pull request by fzaninotto**
## Problem

Adding new buttons or a search engine to the AppBar is cumbersome

## Solution

- [x] Add a `<TitlePortal>` component to avoid copy/pasting a large `<Typography>` from the doc
- [x] Add a `toolbar` prop to let users override the default toolbar
- [x] Add an `alwaysOn` prop to facilitate always on toolbars (and deprecate `component` prop)
- [x] Fix styling glitches
- [x] Fix `<Title>` doesn't work without routing
- [x] Add Storybook
- [x] Add a dedicated `AppBar` doc chapter
- [x] Add a dedicated `Title` doc chapter
- [x] Fix `<AppBar>` usage examples in the doc and demos
- [x] Include screenshots in the doc
- [x] Final proofreading

## Resulting API

To add buttons to the app bar, you can use the `<AppBar>` [`children` prop](#children).

For instance, to add `<ToggleThemeButton>`:

```jsx
// in src/MyAppBar.js
import { 
    AppBar,
    TitlePortal,
    ToggleThemeButton,
    defaultTheme
} from 'react-admin';

const darkTheme = { palette: { mode: 'dark' } };

export const MyAppBar = () => (
    <AppBar>
        <TitlePortal />
        <ToggleThemeButton lightTheme={defaultTheme} darkTheme={darkTheme} />
    </AppBar>
);
```

### CODE REVIEW

File: docs/AppBar.md [hunk 1]
```diff
@@ -68,16 +61,305 @@ const App = () => (
...
+        {/* Your custom appbar content here */}
+    </AppBar>
+);
+```
+
+**Tip**: The `<TitlePortal>` component takes all the available space in the app bar, so it "pushes" the following children to the right.
+
+## `color`
+
+React-admin's `<AppBar>` renders a MUI `<AppBar>`, which supports a `color` prop top set the app bar color depending on the theme. By defaulmt, the app bar color is set to the `secondary` theme color.
```
Comment (djhi) [inline docs/AppBar.md:124]:
```suggestion
React-admin's `<AppBar>` renders a MUI `<AppBar>`, which supports a `color` prop top set the app bar color depending on the theme. By default, the app bar color is set to the `secondary` theme color.
```

File: docs/AppBar.md [hunk 2]
```diff
@@ -68,16 +61,305 @@ const App = () => (
...
+|--------------------------|------------------------------ |
+| `& .RaAppBar-toolbar`    | Applied the main toolbar      |
+| `& .RaAppBar-menuButton` | Applied to the hamburger icon |
+| `& .RaAppBar-title`      | Applied to the title portal   |
+
+To override the style of `<AppBar>` using the [MUI style overrides](https://mui.com/customization/theme-components/), use the `RaAppBar` key.
+
+## `toolbar`
+
+By defautl, the `<AppBar>` renders two buttons in addition to the user menu: the language menu and the refresh button.
```
Comment (djhi) [inline docs/AppBar.md:167]:
```suggestion
By default, the `<AppBar>` renders two buttons in addition to the user menu: the language menu and the refresh button.
```

File: docs/AppBar.md [hunk 3]
```diff
@@ -68,16 +61,305 @@ const App = () => (
...
+        <ToggleThemeButton lightTheme={defaultTheme} darkTheme={darkTheme} />
+    </AppBar>
+);
+```
+
+## `userMenu`
+
+If your app uses [authentication](./Authentication.md), the `<AppBar>` component displays a button to display the user menu on the right side. By default, the user menu only contains a logout button.
+
+The appearance of the user menu depends on the return value of `authProvider.getIdentity()`. The user menu icon renders an anonymous avatar, or the `avatar` property of the identity object if present. If the identity object contains a `fullName` property, it is displayed after the avatar.
```
Comment (djhi) [inline docs/AppBar.md:216]:
```suggestion
The content of the user menu depends on the return value of `authProvider.getIdentity()`. The user menu icon renders an anonymous avatar, or the `avatar` property of the identity object if present. If the identity object contains a `fullName` property, it is displayed after the avatar. 
```

**Review by djhi — CHANGES_REQUESTED** [review]
:muscle:

**Review by djhi — APPROVED** [review]

**Comment by fzaninotto** [comment]
Switching to RFR

**Review by djhi — APPROVED** [review]

File: docs/Reference.md [hunk 4]
```diff
@@ -171,6 +171,8 @@ title: "Index"
 * [`<TextField>`](./TextField.md)
 * [`<TextInput>`](./TextInput.md)
 * [`<TimeInput>`](./TimeInput.md)
+* [<`Title>`](./Title.md)
+* [<`TitlePortal>`](./AppBar.md#children)
```
Comment (cduran-seeri) [inline docs/Reference.md:175]: Thanks! Found a typo, I'm not sure if this is critical
```
<`Title>` to `<Title>`
<`TitlePortal>` to `<TitlePortal>`
```

**Pull request merged**